mycroft/skills/common_query_skill.py

```diff
@@ -80,39 +105,76 @@ def __handle_question_query(self, message):
-                                            "skill_id": self.skill_id,
-                                            "answer": answer,
-                                            "callback_data": callback,
-                                            "conf": confidence}))
+            confidence = self.__calc_confidence(match, search_phrase, level, answer)
+            self.bus.emit(
+                message.response(
+                    {
+                        "phrase": search_phrase,
+                        "skill_id": self.skill_id,
+                        "answer": answer,
+                        "callback_data": callback,
+                        "conf": confidence,
+                    }
+                )
+            )
         else:
             # Signal we are done (can't handle it)
-            self.bus.emit(message.response({"phrase": search_phrase,
-                                            "skill_id": self.skill_id,
-                                            "searching": False}))
-
-    def __calc_confidence(self, match, phrase, level):
+            self.bus.emit(
+                message.response(
+                    {
+                        "phrase": search_phrase,
+                        "skill_id": self.skill_id,
+                        "searching": False,
+                    }
+                )
+            )
+
+    def remove_noise(self, phrase):
+        # remove noise to produce essence
+        phrase = " " + phrase + " "
+        for word in self.translated_noise_words:
+            mtch = " " + word + " "
+            if phrase.find(mtch) > -1:
+                phrase = phrase.replace(word, "")
+        phrase = " ".join(phrase.split())
+        return phrase.strip()
+
+    def __calc_confidence(self, match, phrase, level, answer):
         # Assume the more of the words that get consumed, the better the match
         consumed_pct = len(match.split()) / len(phrase.split())
         if consumed_pct > 1.0:
             consumed_pct = 1.0
+        consumed_pct /= 10
 
         # Add bonus if match has visuals and the device supports them.
-        platform = self.config_core.get('enclosure', {}).get('platform')
+        platform = self.config_core.get("enclosure", {}).get("platform")
         if is_CQSVisualMatchLevel(level) and handles_visuals(platform):
             bonus = 0.1
         else:
             bonus = 0
 
-        if int(level) == int(CQSMatchLevel.EXACT):
-            return 0.9 + (consumed_pct / 10) + bonus
-        elif int(level) == int(CQSMatchLevel.CATEGORY):
-            return 0.6 + (consumed_pct / 10) + bonus
-        elif int(level) == int(CQSMatchLevel.GENERAL):
-            return 0.5 + (consumed_pct / 10) + bonus
-        else:
-            return 0.0  # should never happen
+        # extract topic
+        topic = self.remove_noise(match)
+
+        # calculate relevance
+        matches = 0
+        for word in topic:
```

Comment:
Should this be `for word in topic.split():`? Otherwise it would go per character?

Reply:
Good catch; ty

Reply:
Corrected in latest commit.